msteams-platform/m365-apps/combine-office-add-in-and-teams-app.md

```diff
@@ -0,0 +1,626 @@
+
+### Merge the manifest
+
+The Teams app's manifest is generated at debug-and-sideload time (or build time) from the file manifest.json in the \appPackage folder of the Teams project. *This file is actually a kind of **template** for a manifest. In this article it is referred to as the "template" or "manifest template".* Most of the markup is hardcoded into the template, but there are also some configuration files that contain data that gets injected into the final generated manifest. In this procedure, you do the following tasks:
+
+- Copy markup from the add-in's manifest to the Teams app's manifest template.
+- Edit the configuration files. 
+
+Unless specified otherwise, the file you change is \appPackage\manifest.json.
+
+1. Copy the "$schema" and "manifestVersion" property values from the add-in's manifest to the corresponding properties of the Teams app's manifest template file.
+1. Adjust the "name.full", "description.short", and "description.full" property values as needed to take account of the fact that an Outlook add-in is part of the app. 
+1. Do the same for the "name.short" value. We recommend that you keep the `${{TEAMSFX_ENV}}` on the end of the name. This variable is replaced with "local" when you're debugging on localhost and with "dev" when you're either debugging from a remote domain or in production mode. (Historically, Office Add-in developer documentation uses the term "dev" or "dev mode" to refer to running the add-in on a localhost. It uses the term "prod" or "production mode" to refer to running the add-in on a remote host for staging or production. Teams developer documentation uses the term "local" to refer to running the add-in on a localhost, and the term "dev" to refer to running the add-in on a remote host for remote debugging, which is usually called "staging." We're working on getting the terminology consistent.)
+
+    The following JSON is an example:
+
+    ```
+    "short": "Contoso Tab and Add-in-${{TEAMSFX_ENV}}",
+    ```
+
+    > [!NOTE]
+    > The "name.short" value appears in both the Teams tab capability and the Outlook add-in. Examples: 
+    >
+    > - It is the label under the launch button of the Teams tab.
+    > - It is content of the title bar of the add-in's task pane.
+
+1. If you changed the "name.short" value from its default (which is the name of the project followed by the `${{TEAMSFX_ENV}}` variable), make exactly the same change in all places where the project name appears in the following two files in the root of the project: teamsapp.yml and teamsapp.local.yml.
+1. If there is no "authorization.permissions.resourceSpecific" array in the Teams manifest template, copy it from the add-in manifest as a top-level property. If there already is one in the Teams template, copy any objects from the array in the add-in manifest to the array in the Teams template. The following JSON is an example:
```

Comment:
```suggestion
1. If there's no `authorization.permissions.resourceSpecific` array in the Teams manifest template, copy it from the add-in manifest as a top-level property. If there's already one in the Teams template, copy any objects from the array in the add-in manifest to the array in the Teams template. The following JSON is an example:
```

Reply:
This change is not consistent with the standard practice in Teams docs, which is to put manifest property names in quotation marks.